packages/front-end/components/Settings/EditDataSource/DataSourceMetrics.tsx

```diff
@@ -49,14 +48,21 @@ export default function DataSourceMetrics({
 
   const metrics: MetricInterface[] | undefined = data?.metrics;
 
-  const editMetricsPermissions: { [id: string]: boolean } = {};
+  const editMetricsPermissions: {
+    [id: string]: { canCreate: boolean; canUpdate: boolean };
```

Comment:
We should call is `canDuplicate` instead of `canCreate` since it's confusing why you would create a metric that already exists.